### PR TITLE
Allow ESS to edit non-dynamic entities

### PR DESCRIPTION
**Pull request by sethalves**
- Allow ESS to edit non-dynamic entities
- don't drop localX property edits for entities that aren't in the local tree

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8416/app_12753_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8416/HighFidelity-Beta-PR12753-cfe65409f262dc2b385bfbfa07f785e2bf4cfbbd.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8416/HighFidelity-Beta-PR12753-cfe65409f262dc2b385bfbfa07f785e2bf4cfbbd.exe

**Comment by huffman** [comment]
I put this in last week: https://highfidelity.fogbugz.com/f/cases/13824/Immediately-editing-an-entity-on-the-ESS-after-creating-it-does-not-work

Probably not related because it's not dynamic, but I wanted to mention it.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8426/app_12753_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8426/HighFidelity-Beta-PR12753-1b4287efe7274894c37ed713bdedf5750c1a23f7.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8426/HighFidelity-Beta-PR12753-1b4287efe7274894c37ed713bdedf5750c1a23f7.exe

File: libraries/entities/src/EntityScriptingInterface.cpp [hunk 1]
```diff
@@ -523,6 +527,25 @@ QUuid EntityScriptingInterface::editEntity(QUuid id, const EntityItemProperties&
...
+            }
+            if (properties.localVelocityChanged()) {
+                properties.setVelocity(properties.getLocalVelocity());
+            }
+            if (properties.localAngularVelocityChanged()) {
+                properties.setAngularVelocity(properties.getLocalAngularVelocity());
+            }
+            if (properties.localDimensionsChanged()) {
+                properties.setDimensions(properties.getLocalDimensions());
+            }
```
Comment (AndrewMeadows) [inline libraries/entities/src/EntityScriptingInterface.cpp]: I don't fully understand the full reason/consequence of this change, but let me speculate:

First of all, this `else` case corresponds to the `if (entity)` check, so in the `else` case the entity is not known: we don't know it if has a parent or not.  Whether it has a parent or not at the receiving end changes the behavior...

When the entity doesn't have a parent, then "localPosition" is the same as "position", and for that case this new code will effectively do nothing... except make sure the properties are actually put in the packet (the comment suggests "local" properties are not packed).

When the entity does have a parent then... this change overwrites world-frame properties with local-frame, and the entity will show up near the origin?

Comment (sethalves) [reply libraries/entities/src/EntityScriptingInterface.cpp]: when a script changes localX there's not need to know anything about the parent in order to send the data to the server.  If a script changes X (like position or rotation) it's making a world-frame change and EntityScriptEngine needs to know about the parent in order to send the correct edit request.  Normally, this is taken care of by convertPropertiesFromScriptSemantics, but this is only called if the entity is in the local tree.  If the entity isn't in the local tree, we can still correctly relay localX edits to the server.

I suppose this code could indicate an error if the script attempts to edit a parenting related property in the world frame.

**Comment by sethalves** [comment]
closing in favour of https://github.com/highfidelity/hifi/pull/12763

**Pull request closed without merging**